eq-author/src/App/QuestionnairesPage/QuestionnairesTable/Row.test.js

```diff
@@ -60,51 +60,95 @@ describe("Row", () => {
     expect(focus).toHaveBeenCalled();
   });
 
-  it("should allow duplication of a Questionnaire", () => {
-    const wrapper = shallow(
-      <Row
-        questionnaire={questionnaire}
-        onDeleteQuestionnaire={handleDeleteQuestionnaire}
-        onDuplicateQuestionnaire={handleDuplicateQuestionnaire}
-      />
-    );
+  it("should handle row focus state change correctly", () => {
+    const wrapper = shallow(<Row questionnaire={questionnaire} {...props} />);
+    const tableRow = wrapper.find(TR);
+    // console.log(tableRow.debug());
+    expect(wrapper.state("linkHasFocus")).toBeFalsy();
+    tableRow.simulate("focus");
+    expect(wrapper.state("linkHasFocus")).toBeTruthy();
+    tableRow.simulate("blur");
+    expect(wrapper.state("linkHasFocus")).toBeFalsy();
+  });
 
-    wrapper.find(DuplicateButton).simulate("click");
+  it("should handle button focus state change correctly", () => {
+    const wrapper = shallow(<Row questionnaire={questionnaire} {...props} />);
+    const tableRow = wrapper.find(TR);
+    const actionButton = wrapper.find(TD).find("div");
```

Comment:
This is a fragile selector—the test will break if we change the `div` to a `span` or even add another `div` to the `TD`. Use a `data-test` attribute where possible.

Reply:
Changed this to use data-test instead of looking for the div.